src/ENDFtk/section/GendfBase/src/ctor.hpp

```diff
@@ -0,0 +1,30 @@
+template< typename BufferIterator >
+GendfBase( HeadRecord& head,
+           BufferIterator& begin,
+           const BufferIterator& end,
+           long& lineNumber,
+           int MAT )
+  : Base( head, MAT, head.MF() ),
+    num_legendre_( head.L1() ),
+    num_sigma0_( head.L2() ),
+    lrflag_( head.N1() ),
+    num_groups_( head.N2() ) {
+
+  // populate data records
+  data_ = populateData(head, begin, end, lineNumber, MAT);
```

Comment:
`data_` should be instantiated in the block before the constructor begins.

```suggestion
  : Base( head, MAT, head.MF() ),
    data_(  populateData(head, begin, end, lineNumber, MAT) ),
    num_legendre_( head.L1() ),
    num_sigma0_( head.L2() ),
    lrflag_( head.N1() ),
    num_groups_( head.N2() ) {
```

Reply:
This was an arguably sloppy way of dealing with an issue I had.  `populateData` needs access to `num_legendre_` and others.  I used the member variables inside `populateData`, which means I needed them to be set beforehand.  I could use `head.L1()` instead.  Or I could do the cascading constructor trick that @whaeck is a fan of.  I'll try to see what feels the most natural, as I think all the solutions are about as sloppy as the one I went with originally.